terror/terror.go

```diff
@@ -111,7 +113,12 @@ func (ec ErrClass) EqualClass(err error) bool {
 		return false
 	}
 	if te, ok := e.(*Error); ok {
-		return te.class == ec
+		rfcCode := te.RFCCode()
+		tags := strings.Split(string(rfcCode), ":")
```

Comment:
Prefer to use:
```go
if index := strings.Index(string(rfcCode), ":"); index > 0 {
    if class, has := rfcCode2errClass[string(rfcCode)[:index]]; has {
        return class == ec
    }
}
```
To avoid allocate new slice.